xdsl/ir.py

```diff
@@ -641,8 +641,19 @@ def build(
         """Create a new operation using builders."""
         ...
 
-    def replace_operand(self, operand_idx: int, new_operand: SSAValue) -> None:
+    def replace_operand(self, operand: int | SSAValue,
+                        new_operand: SSAValue) -> None:
         """Replace an operand with another operand."""
+        if isinstance(operand, SSAValue):
+            try:
+                operand_idx = self._operands.index(operand)
+            except ValueError as err:
+                raise ValueError("{} is not an operand of {}!".format(
```

Comment:
```suggestion
                raise ValueError("{} is not an operand of {}".format(
```

Reply:
Why remove the exclamation mark at the end of the error message? I thought it was standard to have them there?

Reply:
Nevermind, just read up on it, seems like no exclamation mark is preferred!